native/cocos/platform/android/java/src/com/cocos/lib/CocosHelper.java

```diff
@@ -395,21 +400,29 @@ public static float[] getSafeArea() {
         return new float[]{0, 0, 0, 0};
     }
     public static void finishActivity() {
-        sActivity.runOnUiThread(new Runnable() {
+        if (GlobalObject.getActivity() == null) {
+            Log.e(TAG, "sActivity is null");
+            return;
+        }
+        GlobalObject.runOnUiThread(new Runnable() {
             @Override
             public void run() {
-                sActivity.finish();
+                GlobalObject.getActivity().finish();
             }
         });
     }
     public static void setKeepScreenOn(boolean keepScreenOn) {
-        sActivity.runOnUiThread(new Runnable() {
+        if (GlobalObject.getActivity() == null) {
+            Log.e(TAG, "sActivity is null");
+            return;
+        }
+        GlobalObject.runOnUiThread(new Runnable() {
             @Override
             public void run() {
                 if (keepScreenOn) {
-                    sActivity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+                    GlobalObject.getActivity().getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
                 } else {
-                    sActivity.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+                    GlobalObject.getActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
```

Comment:
Setting window flags should use `activity`, therefore `GlobalObject.getActivity()` is still needed.